marshal.go

```diff
@@ -393,7 +421,9 @@ func makeResourceObject(v any, vt reflect.Type, m *Marshaler, isRelationship boo
 				continue
 			}
 
+			// relationship marshaler needed for links and name validation
 			rm := new(Marshaler)
+			rm.memberNameValidationMode = m.memberNameValidationMode
```

Comment:
This seems like a potential future bug opportunity ... maybe we move to have a method like `Marshaler.somethingMarshaler` that copies over the things we want?

Reply:
Not sure ... will leave it to you, just seems like having multiple ways that we initialize a marshaler is a good way to introduce bugs. Not a blocking/required change at this point though.

Reply:
Cool. For now I added some helper funcs for creating unmarshalers but I might revisit this topic later